### PR TITLE
added rake task with ability to specify multiple templates to render

### PR DESCRIPTION
**Pull request by aaronwalker**
### Problem

Most projects consist of a number of templates/stacks and wiring these up when generating each template is time consuming and error prone.

### Solution

Adds a rake task that can be included in your projects rake task and you can select the `template`, `extras` files and optionally output it to a file.

### CODE REVIEW

File: lib/cfndsl/rake_task.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+require 'rake'
```
Comment (stevenjack) [inline lib/cfndsl/rake_task.rb]: Thanks for this, really useful task!

It's a little hard to read so I had a little refactor of the class:

```ruby
require "rake"
require "rake/tasklib"
require "cfndsl"

module CfnDsl
  class RakeTask < Rake::TaskLib
    attr_accessor :cfndsl_opts

    def initialize(name = nil)
      yield self if block_given?

      desc "Generate Cloudformation" unless ::Rake.application.last_comment
      task(name || :generate) do |_t, _args|
        cfndsl_opts[:files].each do |opts|
          generate(opts)
        end
      end
    end

    private

    def generate(opts)
      log(opts)
      outputter(opts).puts model(opts[:filename])
    end

    def log(opts)
      type = opts[:output].nil? ? "STDOUT" : opts[:output]
      verbose.puts("Writing to #{type}") if verbose
    end

    def outputter(opts)
      opts[:output].nil? ? STDOUT : file_output(opts[:output])
    end

    def model(filename)
      fail "#{filename} doesn't exist" unless File.exist? filename
      CfnDsl.eval_file_with_extras(filename, extra, verbose).to_json
    end

    def extra
      cfndsl_opts.fetch(:extras, [])
    end

    def verbose
      cfndsl_opts[:verbose] && STDERR
    end

    def file_output(path)
      File.open(File.expand_path(path), "w")
    end
  end
end
```

What do you reckon?

File: README.md [hunk 2]
```diff
@@ -215,7 +215,29 @@ would generate a template with 5 instances declared.
...
+    files: [{
+      filename: 'templates/application.rb',
+      output: 'application.json'
+    }],
+    extras: [{
+      yaml: 'templates/default_params.yml'
+    }]
+  }
+end
+```
```
Comment (stevenjack) [inline README.md]: I think we need a quick explanation that `output` is optional and how you can use the `verbose` key.

Comment (stevenjack) [inline README.md]: I think we need a quick explanation that `output` is optional and how you can use the `verbose` key.

**Comment by stevenjack** [comment]
@aaronwalker This is great, been thinking (Not for this PR) but it could be better if you were able to define a `.cfndsl.yml` config in your project directory which could look like:

```yaml

extras: "path/to/default/extras.yml"
templates:
  - 
    filename: "component_one/template.json"
    extras: "component_one/extras.yml"
    output: "stacks/component_one.json"
```

You then just run `cfndsl` and it'll generate the templates based on this configuration (Bonus of having default extras at the top then each template can override and have it's own extras).

File: lib/cfndsl/rake_task.rb (hunk 1 above)
Comment (aaronwalker) [inline lib/cfndsl/rake_task.rb]: @stevenjack looks good...I am happy to update the PR with these changes. Or do you want me to close this PR and create a new one?

**Comment by aaronwalker** [comment]
@stevenjack yeah like to proposal for rendering templates via a config file. I will create a issue to track this outside of the PR as I have a few other idea I'd like to put forward around this.

**Comment by aaronwalker** [comment]
closing will create a new PR with just the required changes

**Pull request closed without merging**